mesheryctl/internal/cli/root/system/logs.go

```diff
@@ -212,27 +244,45 @@ mesheryctl system logs meshery-istio
 					}
 					defer logs.Close()
 
-					buf := new(bytes.Buffer)
-					_, err = io.Copy(buf, logs)
-					if err != nil {
-						return fmt.Errorf("error in copy information from logs to buf")
-					}
-
-					// Append this to data to be printed
-					for _, str := range strings.Split(buf.String(), "\n") {
-						data = append(data, fmt.Sprintf("%s\t|\t%s", name, str))
+					if follow {
+						go func() {
+							for {
+								buf := make([]byte, BYTE_SIZE)
+								numBytes, err := logs.Read(buf)
+								if numBytes == 0 {
+									continue
+								}
+								if err == io.EOF {
+									break
+								}
+								if err != nil {
+									break
```

Comment:
Here in the case of unknown error, you can throw the error to the client.

Reply:
@Revolyssup how to throw the error to the client from go routine? 

I saw we can use **errgroup** to group go routine and handle errors.

Reply:
Yeah you can use that as well. Thats usually used when a lot of synchronisation is needed.
Although here it feels like goroutine's are not really required. There isn't much synchronisation required and this is added complexitiy.

So currently you start a consumer blocking go routine and [ producer blocking go routine only when follow is present]. Then you block the parent function until producer returns and then you close the channel. Or in case of of non follow you immediately close the channel. 

A much simpler solution is:
Dont use any go routines here. Check if `!follow` then print out the outputs and return immediately. 
Else block the main function in a for loop and keep printing until EOF.  No extra synchronisation overhead needed here.